pyasdf/block.py

```diff
@@ -521,6 +528,46 @@ def compression(self, compression):
     def is_compressed(self):
         return self._compression is not None
 
+    @property
+    def checksum(self):
+        return self._checksum
+
+    def _set_checksum(self, checksum):
+        if checksum == b'\0' * 16:
+            self._checksum = None
+        else:
+            self._checksum = checksum
+
+    def _calculate_checksum(self, data):
+        m = hashlib.new('md5')
+        m.update(self.data)
```

Comment:
I haven't super kept on the details of how arrays are written, but I wonder if this could avoid an extra loop over all the data if instead it were performed in chunks while simultaneously streaming the data block to disk.  Of course this would only work for a random access write since you'd have to go back to write the checksum.  But for typical writes to disk I wonder if there's room for improvement here--compute the checksum on some block of the data while it's still paged in for writing, then be done with it for good.